README.md

```diff
@@ -1,3 +1,17 @@
 # SNARK Verifier
 
 Generic (S)NARK verifier.
+
+## User Guide
+
+This project is tailored for verifying [Halo2](https://github.com/privacy-scaling-explorations/halo2)-generated proofs on Ethereum.
+
+Additionally, within this repository, there is an example provided for verifying an aggregated proof, which serves as proof for multiple other proofs. You can test it by executing `cargo run --example evm-verifier-with-accumulator`.
+
+As this effort is continuously evolving, there are multiple repository variants to choose from. For optimal results:
+
+- For production use, it is recommended to use [axiom’s fork](https://github.com/axiom-crypto/snark-verifier), which has been audited.
+- In the future, consider using [halo2-solidity-verifier](https://github.com/privacy-scaling-explorations/halo2-solidity-verifier). However, please note the following:
```

Comment:
This section seems only applicable for solidity verifier generation, we still need to use `snark-verifier` for proof aggregation even `halo2-solidity-verifier` is mature.

```suggestion
- In the future, consider using [halo2-solidity-verifier](https://github.com/privacy-scaling-explorations/halo2-solidity-verifier) for solidity verifier generation. However, please note the following:
```